ui/test/unit/registration/controllers/patientCommonController.spec.js

```diff
@@ -76,6 +76,20 @@ describe('PatientCommonController', function () {
         expect(scope.confirmationPrompt).not.toHaveBeenCalled();
     });
 
+it('checks that the confirmation popup is not prompted on the Registration second page when the home button is clicked and the config is enabled', function () {
+        scope.showSaveConfirmDialogConfig = true;
+        $state.current.name = "patient.visit";
+        scope.onHomeNavigate = jasmine.createSpy("onHomeNavigate");
+        scope.confirmationPrompt = jasmine.createSpy("confirmationPrompt");
+        var element = angular.element("<a ng-click='onHomeNavigate()'>");
+        var compiled = $compile(element)(scope);
+        compiled.triggerHandler('click');
+        expect(scope.onHomeNavigate).toHaveBeenCalled();
+        expect(scope.showSaveConfirmDialogConfig).toBe(true);
+        expect($state.current.name).toBe("patient.visit");
+        expect(scope.confirmationPrompt).not.toHaveBeenCalled();
+    });
```

Comment:
OK that's great but can you also demonstrate that the popup does **not** show up if `$state.current.name` is not `"patient.visit"`? Because that is precisely the behaviour that was spotted during QA.

Reply:
@krishnanspace I meant that we would need one specific test case method that shows that the popup is not activated when `$state.current.name` is not `"patient.visit"`. Does that make sense?

I'm asking this because that's the exact bug that was spotted during QA.

Reply:
@mks-d It will show the pop-up for all other states except patient.visit but wont show it on patient.visit state. Earlier it was found out that the prompt was appearing in patient.visit too.

Reply:
I mean the pop-up would be activated when the $state.current.name is not patient.visit

Reply:
Please note that state.current.name is set to patient.visit only for the visit details (registration second page). From the card (BAH-497), its understood that the confirmation is to be shown only for the first page (patient edit page). The code looks ok for that.